pal/src/tickcounter_sl.c

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include "azure_c_shared_utility/gballoc.h"
+
+#include <stdint.h>
+#include <time.h>
+#include "azure_c_shared_utility/tickcounter.h"
+#include "azure_c_shared_utility/optimize_size.h"
+#include "azure_c_shared_utility/xlogging.h"
+
+#define INVALID_TIME_VALUE      (time_t)(-1)
+typedef struct TICK_COUNTER_INSTANCE_TAG
+{
+    time_t init_time_value;
+    tickcounter_ms_t current_ms;
+} TICK_COUNTER_INSTANCE;
+
+TICK_COUNTER_HANDLE tickcounter_create(void)
+{
+    struct timespec tspec;
+    TICK_COUNTER_INSTANCE* result = (TICK_COUNTER_INSTANCE*)malloc(sizeof(TICK_COUNTER_INSTANCE));
+    if (result != NULL)
+    {
+        clock_gettime(CLOCK_REALTIME, &tspec);
```

Comment:
>CLOCK_REALTIME [](start = 22, length = 14)

Is `CLOCK_MONOTONIC` available? If so, that should be used rather than `CLOCK_REALTIME`.